### PR TITLE
Field context

### PR DESCRIPTION
**Pull request by AlexJukes**
Adds the ability to pass a top level value to all fields for use when customMapping.

Useful for a case like in AMUI, where we have some top level state which we want to use to simultaneously update all fields, without having to pass it individually to each one.

### CODE REVIEW

File: lib/generateMappings.jsx [hunk 1]
```diff
@@ -10,17 +10,17 @@ const generateMappings = ({
...
-  if (mappings.default && typeof mappings.default === 'function') {
-    return mappings[type]
-      ? mappings[type]({
-        key, onChange, onBlur, setFieldValue, ...field,
-      })
-      : mappings.default({
-        key, onChange, onBlur, setFieldValue, ...field,
-      });
+}, fieldContext) => {
+  if (!mappings.default || typeof mappings.default !== 'function') {
```
Comment (AlexJukes) [inline lib/generateMappings.jsx]: This is mainly an aesthetic refactor. I prefer returning the edge case immediately, and then having the main `return` value of the function being the one we usually expect.

File: test/unit/rolling-fields.test.jsx [hunk 2]
```diff
@@ -211,6 +211,29 @@ describe('Rolling fields', () => {
...
   });
 
+  it('will use the provided field context when mapping components', () => {
+    const placeHolderText = 'Place me in my holder!';
+
+    const fields = [
+      { name: 'test', type: 'custom' },
+    ];
+    const mappings = {
+      custom: ({ key, name }, fieldContext) => (
```
Comment (sithmel) [inline test/unit/rolling-fields.test.jsx]: Why mixing up destructuring and positional arguments?
```js
custom: ({ key, name, fieldContext }) => ( ...
```

**Comment by sithmel** [comment]
This package needs updating for security vulns. We can do it here, or in a different PR

File: test/unit/rolling-fields.test.jsx (hunk 2 above)
Comment (AlexJukes) [reply test/unit/rolling-fields.test.jsx]: @sithmel the thinking was that the first argument is all the props specifically passed to the field, whereas the second argument would be almost like an optional, global "config" context for the field. If you think it's clearer though, we can move it into the single argument object. Any opinions on this @danielreibl

Comment (sithmel) [reply test/unit/rolling-fields.test.jsx]: The way you implemented it, you will be forced to do
```js
custom: ({ key, name }, fieldContext, somethingElse, anotherOne) => (
```
or am I missing something ?

Comment (danielreibl) [reply test/unit/rolling-fields.test.jsx]: We use the spread operator with the first parameter, so if we move it in there we also need to change that otherwise, it gets applied as a prop to the inputs

Comment (danielreibl) [reply test/unit/rolling-fields.test.jsx]: Or we need to destructure this property even in places where we don't want to use it's value

Comment (AlexJukes) [reply test/unit/rolling-fields.test.jsx]: @sithmel @danielreibl makes a good point, this is the reason we would want to keep the field values and additional, 'global' values separated. If you're ok with it, i will merge this in
